base/decfp.jl

```diff
@@ -0,0 +1,32 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+# These define the types for IEEE 754-2008 Decimal Floating Point formats
+# Support for these types is available via packages, such as DecFP.jl
+
+export AbstractFlt, BinaryFloat, DecimalFloat, DecimalFloatBinary, DecimalFloatPacked
+
+"""Abstract types for all floating point types"""
+abstract AbstractFlt{b} <: Real
```

Comment:
As I said in #14241, I'd tried to use `AbstractFloat{b}`, as @StefanKarpinski had suggested, but it got an error.  If somebody knows how to get past that error, I'd also prefer to have it that way.
If you prefer, in the meantime, I can just remove those, and simply add DecimalFloat and the two subtypes for binary and packed formats.